src/uu/uniq/src/uniq.rs

```diff
@@ -253,8 +255,89 @@ fn get_long_usage() -> String {
     )
 }
 
+fn scrub_legacy(x: &OsString, flags_with_value: &HashSet<char>) -> Option<OsString> {
+    let stringed: Vec<char> = x.to_string_lossy().chars().collect();
+
+    if stringed.len() <= 1 {
+        return Some(x.to_owned());
+    }
+
+    if stringed[0] == '-' {
```

Comment:
To illustrate what I meant, this could be roughly:
```rust
if x.starts_with("--") {
    Some(x)   
} else if let Some(rest) = x.strip_prefix('-') {
    if rest.iter().all(char::is_numeric) {
         Some(rest)
    } else {
         None
    }
} else if let Some(rest) = x.strip_prefix('+') {
   // same here
}
```

Another thing you could do is make this a `is_legacy_argument` function and split the arguments based on that:

```rust
let (legacy, rest): (Vec<_>, Vec<_>) = args.iter().partition(is_legacy);
```

`is_legacy` then doesn't need to return the `OsString`.

Reply:
cool, i didn't realize it could be done by matching on the assignement, i'm gonna implement it when i get back home, still i think the `partition` solution wouldn't work alone because it would also need to split strings like `-5s3` into `-5` and `-s 3`